File: DailyChallenges/CPP/LeetCode_2348.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long ans = 0, dp0 = 1, dp1 = 0;
        for (int num: nums){
            if (num == 0){
                dp1 += 1;
                dp0 += dp1-1;
            } else {
                ans += dp0 + dp1 -1;
                dp0 = 1;
                dp1 = 0;
            }
        } 
        return ans + dp0 + dp1 - 1;
    }
};

File: DailyChallenges/Scala/LeetCode_2348.scala
object Solution {
    def zeroFilledSubarray(nums: Array[Int]): Long = {
        def helper(i: Int, cnt: Long, lc: Long): Long = if (i == nums.length) cnt else {
            if (nums(i) == 0) helper(i+1, cnt+lc+1, lc+1) else helper(i+1, cnt, 0)
        }
        helper(0, 0, 0)
    }
}
